tensorflow/compiler/xla/service/heap_simulator.h

```diff
@@ -444,6 +456,25 @@ class GlobalDecreasingSizeBestFitHeap : public HeapAlgorithm<BufferType> {
       const BufferInterval& interval) const;
 };
 
+// This class implements an algorithm that will output multiple heaps. Each heap
+// size is constrained by a given limit. Note that the constraint is soft,
+// meaning that valid heap results are generated even if there are some buffer
+// sizes larger than the given constraint size.
+class ConstrainedGlobalDecreasingSizeBestFitHeap
```

Comment:
Could you write the pseudocode description of the algorithm as well?

Reply:
No problem. Will add.

Reply:
Added.